src/modules/job-info/job_state.c

```diff
@@ -214,6 +297,10 @@ void job_state_destroy (void *data)
             zhashx_destroy (&jsctx->index);
         if (jsctx->transitions)
             zlistx_destroy (&jsctx->transitions);
+        if (jsctx->events)
+            flux_future_destroy (jsctx->events);
+        if (jsctx->events_reconnect)
+            flux_watcher_destroy (jsctx->events_reconnect);
```

Comment:
Test for non-NULL is unnecessary for both of these destructors (and generally for all public flux API functions - it's a bug if we've missed any)